mm2src/mm2_main/src/lp_swap/maker_swap_v2.rs

```diff
@@ -463,13 +815,34 @@ impl<MakerCoin: CoinAssocTypes, TakerCoin: CoinAssocTypes> NegotiationData<Maker
     fn to_stored_data(&self) -> StoredNegotiationData {
         StoredNegotiationData {
             taker_payment_locktime: self.taker_payment_locktime,
+            taker_funding_locktime: self.taker_funding_locktime,
             maker_coin_htlc_pub_from_taker: self.maker_coin_htlc_pub_from_taker.to_bytes().into(),
             taker_coin_htlc_pub_from_taker: self.taker_coin_htlc_pub_from_taker.to_bytes().into(),
             maker_coin_swap_contract: self.maker_coin_swap_contract.clone().map(|b| b.into()),
             taker_coin_swap_contract: self.taker_coin_swap_contract.clone().map(|b| b.into()),
             taker_secret_hash: self.taker_secret_hash.clone().into(),
         }
     }
+
+    fn from_stored_data(
+        stored: StoredNegotiationData,
+        maker_coin: &MakerCoin,
+        taker_coin: &TakerCoin,
+    ) -> Result<Self, MmError<SwapRecreateError>> {
+        Ok(NegotiationData {
+            taker_payment_locktime: stored.taker_payment_locktime,
+            taker_funding_locktime: stored.taker_funding_locktime,
+            maker_coin_htlc_pub_from_taker: maker_coin
+                .parse_pubkey(&stored.maker_coin_htlc_pub_from_taker.0)
+                .map_err(|e| SwapRecreateError::FailedToParseData(e.to_string()))?,
+            taker_coin_htlc_pub_from_taker: taker_coin
+                .parse_pubkey(&stored.taker_coin_htlc_pub_from_taker.0)
+                .map_err(|e| SwapRecreateError::FailedToParseData(e.to_string()))?,
+            maker_coin_swap_contract: None,
+            taker_coin_swap_contract: None,
```

Comment:
These are `None` here but retrieved in the mirror implementation for taker swap similar to `p2p_keypair`.